### ISSUE TITLE
API header file for ml_kem and ml_dsa seems to contains the wrong function names

### ISSUE THREAD

**Issue by matteo-cristino** [open-quantum-safe/liboqs#1767]
Trying to include your ml_kem and ml_dsa library in our code we got an error sayign the function contained in the api.h file were not defined. Lookin closer to the files:
* `config.h` and `sign.h` for ml_dsa
* `paramsh.h` and `kem.h` for ml_kem

It seems that the functions in api.h file have the wrong name
* `dilithium2`, `dilithium3` and `dilithium5` should be replace respectively with `ml_dsa_44_ipd`, `ml_dsa_65_ipd` and `ml_dsa_87_ipd`
* `kyber512`, `kyber768` and `kyber1024` should be replace respectively with `ml_kem_512_ipd`, `ml_kem_768_ipd` and `ml_kem_1024_ipd`

**Comment by SWilson4** [comment]
Hi @matteo-cristino, thanks for the report. It sounds to me like you might be including a non-public header file. You should not be including the `.h` files contained in the `ml_dsa` and `ml_kem` folders directly. Instead, you can include `<oqs/sig.h>` or `<oqs/kem.h>`, or just `<oqs/oqs.h>` if you want access to the entire `liboqs` API. Please see [this example for signature use](https://github.com/open-quantum-safe/liboqs/blob/2bb1d2584c275a861d2f763e2d16200541bc6f9d/tests/example_sig.c).

Please let me know if that fixes the problem.

**Comment by jaromil** [comment]
Thanks for your prompt answer @SWilson4 ! Our situation is different, since we include only **some** liboqs (and pqclean) implementations as static libraries inside our software, which is an isolated VM for secure execution with zero dependencies. In this specific case @matteo-cristino solved it like this https://github.com/dyne/Zenroom/pull/850/commits/df2fa4352de6298cfe8c5693eeca6ab4b1f895bd 

In conclusion: it is of course fine if you have different names in api.h for reasons inherent to liboqs design, we may solve it even without modifying your header and using `extern` directives in our own code, to ease future update.

You may close this issue; it is solved for us.
